src/cargo/core/manifest.rs

```diff
@@ -28,13 +29,15 @@ pub struct Manifest {
     profiles: Profiles,
     publish: bool,
     replace: Vec<(PackageIdSpec, Dependency)>,
+    augment: HashMap<Url, Vec<Dependency>>,
```

Comment:
Hm, specifying `augment` as a `Dependency` (that is, a constraint on possible versions) does not feel semantically right. What does it mean to augment a registry with `foo 1.0.*`?

Should we add a constraint that you must explicitelly add a `=x.y.z` constraint to the augment? This seems better from the theoretical pov, but I don't see any real benefits except perhaps for the better readability:

 ```toml
 [augment.crates-io]
 xml-rs = { git = "https://github.com/aturon/xml-rs", branch = "servo-fix" }
```

vs

 ```toml
 [augment.crates-io]
 xml-rs = { version = "0.9.2", git = "https://github.com/aturon/xml-rs", branch = "servo-fix" }
```

That is, I am a bit unsure about

> Cargo will unconditionally resolve all entries in the [augment] section to precise dependencies, encoding them all in the lock file whether they're used or not.

section of the RFC. Should cargo really *resolve* entries in augment? What if it finds several versions of `[augment]` (it's not possible with `git` source, but may be possible with some theoretical other), which one should it choose?